### PR TITLE
ref: prep repo for move to @sentry-javascript

### PR DESCRIPTION
**Pull request by JonasBa**
- [x] port eslint config and fix errors
- [x] output to build
- [x] cleanup esm build artifacts
- [x] port tsconfig
- [x] use yarn1.x instead of npm
- [x] downgrade ts to match sentry-javascript repo

### CODE REVIEW

File: src/hubextensions.ts [hunk 1]
```diff
@@ -17,6 +17,9 @@ type StartTransaction = (
 
 // Takes a transaction and determines if it should be profiled or not. If it should be profiled, it returns the
 // profile_id, otherwise returns undefined. Takes care of setting profile context on transaction as well
+/**
+ *
+ */
```
Comment (anonrig) [inline src/hubextensions.ts:22]:
```suggestion
```

File: src/hubextensions.ts [hunk 2]
```diff
@@ -142,6 +145,9 @@ export function stopTransactionProfile(
 // Wraps startTransaction and stopTransaction with profiling related logic.
 // startProfiling is called after the call to startTransaction in order to avoid our own code from
 // being profiled. Because of that same reason, stopProfiling is called before the call to stopTransaction.
+/**
+ *
+ */
```
Comment (anonrig) [inline src/hubextensions.ts:150]:
```suggestion
```

File: src/integration.ts [hunk 3]
```diff
@@ -38,11 +38,24 @@ function addToProfileQueue(profile: RawThreadCpuProfile): void {
 // and inspect each event to see if it is a transaction event and if that transaction event
 // contains a profile on it's metadata. If that is the case, we create a profiling event envelope
 // and delete the profile from the transaction metadata.
+/**
+ *
+ */
```
Comment (anonrig) [inline src/integration.ts:43]: There are others as well.

File: src/integration.ts [hunk 4]
```diff
@@ -109,16 +124,17 @@ export class ProfilingIntegration implements Integration {
 
         const profilesToAddToEnvelope: Profile[] = [];
 
-        for (let i = 0; i < profiledTransactionEvents.length; i++) {
-          const profiledTransaction = profiledTransactionEvents[i];
-          const profile_id = profiledTransaction?.contexts?.['profile']?.['profile_id'];
+        for (const profiledTransaction of profiledTransactionEvents) {
+          const profileContext = profiledTransaction.contexts && profiledTransaction.contexts['profile']
```
Comment (anonrig) [inline src/integration.ts:128]: I think we should revert these changes. `node-profiling` supports Node.js 14 and up, which supports optional chaining: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Comment (JonasBa) [reply src/integration.ts:128]: This is actually a rule in the tsconfig and afaik we do this because of bundle size constraints (optional chaining doesn't minify well)

File: src/hubextensions.ts (hunk 1 above)
Comment (JonasBa) [reply src/hubextensions.ts:22]: Will check all of these and add relevant comments, was just to satisfy the linters for now

File: src/integration.ts (hunk 4 above)
Comment (timfish) [reply src/integration.ts:128]: We disable these lints in `@sentry/node` though because bundle size is less of a concern there:
https://github.com/getsentry/sentry-javascript/blob/develop/packages/node/.eslintrc.js

Comment (JonasBa) [reply src/integration.ts:127]: Aghh, nice find @timfish, I missed that! Will revert these changes

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+module.exports = {
```
Comment (JonasBa) [inline .eslintrc.js:1]: Once moved, this will extend from the root eslintrc in @sentry-javascript. The only thing we'll need to keep is a couple of ignore patterns and rules

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -25,8 +25,6 @@ jobs:
           # distributions and container images.
 
           # linux x64 glibc
-          - os: ubuntu-20.04
```
Comment (JonasBa) [inline .github/workflows/build.yml:28]: Once merged to sentry-javascript we'll adopt their v7.xx version so it's also a good time to drop node 14.

**Review by anonrig — APPROVED** [review]
